src/Assets/js/bot/add.js

```diff
@@ -0,0 +1,30 @@
+function isInt(value) {
+    return !isNaN(value) &&
+        parseInt(Number(value)) == value &&
+        !isNaN(parseInt(value, 10));
+}
+
+function isSnowflake(value) {
+    return isInt(value) && value.length >= 16;
+}
+
+document.getElementById('add').addEventListener('submit', function (e) {
+    var value = document.getElementById('botid');
+    if (!value) {
+        var inputs = document.querySelectorAll('input[type=checkbox]:checked');
+        if (inputs.length == 0) {
```

Comment:
This doesn't seem right for checking if lists are selected, as there are also checkboxes in the main form itself that might be checked? Maybe give all the list checkboxes an id like `list-${list.name}`, and then you can use `input[type=checkbox][id^=list-]:checked` here